codegenerator/cli/templates/static/codegen/src/eventFetching/PartitionedFetchState.res

```diff
@@ -22,181 +17,92 @@ let make = (
-
-  let partitions = []
-
-  if numAddresses <= maxAddrInPartition {
-    let partition = FetchState.make(
-      ~partitionId=partitions->Array.length,
-      ~staticContracts,
-      ~dynamicContractRegistrations,
-      ~startBlock,
-      ~logger,
-      ~isFetchingAtHead=false,
-    )
-    partitions->Js.Array2.push(partition)->ignore
-  } else {
-    let staticContractsClone = staticContracts->Array.copy
-
-    //Chunk static contract addresses (clone) until it is under the size of 1 partition
-    while staticContractsClone->Array.length > maxAddrInPartition {
-      let staticContractsChunk =
-        staticContractsClone->Js.Array2.removeCountInPlace(~pos=0, ~count=maxAddrInPartition)
-
-      let staticContractPartition = FetchState.make(
-        ~partitionId=partitions->Array.length,
-        ~staticContracts=staticContractsChunk,
-        ~dynamicContractRegistrations=[],
-        ~startBlock,
-        ~logger,
-        ~isFetchingAtHead=false,
-      )
-      partitions->Js.Array2.push(staticContractPartition)->ignore
-    }
-
-    let dynamicContractRegistrationsClone = dynamicContractRegistrations->Array.copy
-
-    //Add the rest of the static addresses filling the remainder of the partition with dynamic contract
-    //registrations
-    let remainingStaticContractsWithDynamicPartition = FetchState.make(
-      ~partitionId=partitions->Array.length,
-      ~staticContracts=staticContractsClone,
-      ~dynamicContractRegistrations=dynamicContractRegistrationsClone->Js.Array2.removeCountInPlace(
-        ~pos=0,
-        ~count=maxAddrInPartition - staticContractsClone->Array.length,
-      ),
-      ~startBlock,
-      ~logger,
-      ~isFetchingAtHead=false,
-    )
-    partitions->Js.Array2.push(remainingStaticContractsWithDynamicPartition)->ignore
-
-    //Make partitions with all remaining dynamic contract registrations
-    while dynamicContractRegistrationsClone->Array.length > 0 {
-      let dynamicContractRegistrationsChunk =
-        dynamicContractRegistrationsClone->Js.Array2.removeCountInPlace(
-          ~pos=0,
-          ~count=maxAddrInPartition,
-        )
-
-      let dynamicContractPartition = FetchState.make(
-        ~partitionId=partitions->Array.length,
-        ~staticContracts=[],
-        ~dynamicContractRegistrations=dynamicContractRegistrationsChunk,
-        ~startBlock,
-        ~logger,
-        ~isFetchingAtHead=false,
-      )
-      partitions->Js.Array2.push(dynamicContractPartition)->ignore
-    }
-  }
-
-  if Env.Benchmark.shouldSaveData {
-    Benchmark.addSummaryData(
-      ~group="Other",
-      ~label="Num partitions",
-      ~value=partitions->Array.length->Int.toFloat,
-    )
-  }
+  let partition = FetchState.make(
+    ~staticContracts,
+    ~dynamicContracts=dynamicContractRegistrations,
+    ~startBlock,
+    ~maxAddrInPartition,
+    ~isFetchingAtHead=false,
+  )
 
   {
     maxAddrInPartition,
-    partitions,
+    partitions: [partition],
```

Comment:
This is temporary right now right?

Reply:
Yes, the PartitionedFetchState is now 100% useless, I didn't remove it in the PR only to minimize already huge diff.